src/commands/analyze/frameworks.ts

```diff
@@ -35,15 +39,95 @@ export async function isTypescript(contents: Record<string, string>): Promise<bo
     return false;
 }
 
-export async function getEntryfile(contents: Record<string, string>): Promise<string> {
+export async function findEntryFile(
+    componentPath: string,
+    contents: Record<string, string>,
+    patterns: RegExp[],
+    defaultFile: string,
+): Promise<string> {
+    let entryFile = await getEntryFileFromPackageJson(componentPath, contents, patterns);
+
+    if (!entryFile) {
+        entryFile = await findFileByPatterns(componentPath, patterns, EXTENSIONS);
+    }
+
+    // If no entry file is found, use the default
+    return entryFile || defaultFile;
+}
+
+async function findFileByPatterns(
+    directory: string,
+    patterns: RegExp[],
+    extensions: string[],
+): Promise<string | undefined> {
+    const entries = await fs.readdir(directory, { withFileTypes: true });
+
+    const contents: Record<string, string> = {};
+
+    // Populate contents with package.json and any files needed for direct access
+    for (const entry of entries) {
+        const fullPath = path.join(directory, entry.name);
+        if (entry.isFile()) {
+            const content = await fs.readFile(fullPath, "utf-8");
+            contents[entry.name] = content;
+        }
+    }
+
+    for (const entry of entries) {
+        const fullPath = path.join(directory, entry.name);
+
+        if (entry.isDirectory()) {
+            // Skip excluded directories
+            if (EXCLUDED_DIRECTORIES.includes(entry.name)) continue;
+
+            // Recursively search within subdirectories
+            const result = await findFileByPatterns(fullPath, patterns, extensions);
+            if (result) return result;
+        } else if (entry.isFile() && extensions.some((ext) => entry.name.endsWith(ext))) {
+            // Check if the file content matches all given patterns
+            const content = await fs.readFile(fullPath, "utf-8");
```

Comment:
Files are read twice: once during the initial population of `contents`, and again when pattern matching. Instead, we could leverage `contents` to store file contents on the first read and reuse them for pattern matching

```
        if (!contents[entry.name]) {
            contents[entry.name] = await fs.readFile(fullPath, "utf-8");
        }
        const content = contents[entry.name];
```

Reply:
Great suggestion - thanks